### PR TITLE
Fill out more of the node:util implementation

### PR DESCRIPTION
**Pull request by jasnell**
Handful of additional `node:util` APIs

### CODE REVIEW

File: src/node/util.ts [hunk 1]
```diff
@@ -183,18 +259,42 @@ export default {
...
-// getSystemErrorMap/getSystemErrorName -- libuv specific. No use in workerd?
-// is{Type} variants -- these are deprecated in Node. Use util.types
-// toUSVString -- Not clear how broadly this is used in the ecosystem outside of node.js.
-//                also this is soon to be obsoleted by toWellFormed in the language.
-// transferableAbortSignal/transferableAbortController -- postMessage and worker threads
-//      are not implemented in workerd. No use case for these.
+//   * util._errnoException
+//   * util._exceptionWithHostPort
+//   * util.getSystemErrorMap
+//   * util.getSystemErrorName
```
Comment (jasnell) [inline src/node/util.ts:284]: The four above here relate closely to the `uv` module. There's likely no valid use case for these in workers outside of Just Make It Work compat.

File: src/node/util.ts [hunk 2]
```diff
@@ -183,18 +259,42 @@ export default {
...
+//   * util.isFunction
+//   * util.isNull
+//   * util.isNullOrUndefined
+//   * util.isNumber
+//   * util.isObject
+//   * util.isPrimitive
+//   * util.isRegExp
+//   * util.isString
+//   * util.isSymbol
+//   * util.isUndefined
```
Comment (jasnell) [inline src/node/util.ts:300]: These are all deprecated. I think we should only add them if someone specifically asks for it or we find a module we want to support that requires them.

**Comment by fhanau** [comment]
Code looks good to me. Some observations:
- util.log() is deprecated.
- util.toUSVString is doc-only deprecated as of https://github.com/nodejs/node/pull/49725. Not sure if we want to keep those, but since it's not marked as deprecated in the main documentation we can implement it.
- deprecations.md in the Node docs also lists util.debug() as deprecated, but that's not reflected in https://nodejs.org/api/util.html#utildebugsection so adding it makes sense to me.
- deprecate() is a no-op as implemented here, glad we're having a TODO for it.
- Are we adding tests for this?

**Comment by jasnell** [comment]
`util.log()`, `util.toUSVString()` are both still fairly common in the wild, so even with deprecation they make sense.

I'll see about adding some tests.

**Review by fhanau — APPROVED** [review]
Tests LGTM, having more would be nice to have but the attack surface of these functions is small so not required. Based on the number of commits consider combining some of them/using squash-merge.

**Pull request merged**